examples/pixelcnn/train.py

```diff
@@ -168,40 +133,51 @@ def _prepare(x):
   return jax.tree_map(_prepare, xs)
 
 
-def restore_checkpoint(optimizer, ema):
-  return checkpoints.restore_checkpoint(FLAGS.model_dir, (optimizer, ema))
+def restore_checkpoint(model_dir, optimizer, ema):
+  return checkpoints.restore_checkpoint(model_dir, (optimizer, ema))
 
 
-def save_checkpoint(optimizer, ema):
+def save_checkpoint(model_dir, optimizer, ema):
   # get train state from the first replica
   optimizer, ema = jax.device_get(
       jax.tree_map(lambda x: x[0], (optimizer, ema)))
   step = int(optimizer.state.step)
-  checkpoints.save_checkpoint(FLAGS.model_dir, (optimizer, ema), step, keep=3)
+  checkpoints.save_checkpoint(model_dir, (optimizer, ema), step, keep=3)
+
 
+def train_and_evaluate(config: ml_collections.ConfigDict, model_dir: str):
+  """Executes model training and evaluation loop.
 
-def train(pcnn_module, model_dir, batch_size, init_batch_size, num_epochs,
-          learning_rate, decay_rate, run_seed=0):
-  """Train model."""
+  Args:
+    config: Hyperparameter configuration for training and evaluation.
+    model_dir: Directory to store model data.
+  """
   if jax.host_count() > 1:
     raise ValueError('PixelCNN++ example should not be run on more than 1 host'
                      ' (for now)')
 
+  # Make sure tf does not allocate gpu memory.
+  tf.config.experimental.set_visible_devices([], 'GPU')
```

Comment:
Could we instead move this to the `main()` and `test.setUp()`?
See #472

Reply:
Done made the changes, thanks for pointing out. 

Just curious, why can't we have this in `train_and_evaluate`? Since the unit tests, benchmark and even the main would just call that method.

Reply:
We could also have it here, but I think it's cleaner to have this kind of "setup code" in the calling functions.
Don't benchmarks usually call the `main()` function?